### DISCUSSION TITLE
repo name for Definition similar to @repository

### DISCUSSION THREAD

**Discussion by CHB-BH** [dagster-io/dagster#18390] [Q&A] [answered]
Hi Dagster Team,

Is it possible to add `repo_name` to a Definition? e.g. in `@repository`, we can do

```python
@repository(name="my_repo_name"):
    return [......]
```

Is it possible to do something like this?

```python
Definition(name="my_repo_name", assets=[......])
```

Will `@repository` be deprecated in the future, or can we continue to use `@repository`?

Thanks.

**Comment by smackesey** [comment, marked as answer]
Hi, it is not possible because the `name` argument of `@repository` was designed for distinguishing multiple repositories within a code location, but there can be only one `Definitions` per code location. This always receives a default name.

>Will @repository be deprecated in the future.

Yes. You should switch to `Definitions` if possible.
